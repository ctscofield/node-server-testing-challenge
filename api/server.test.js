const request = require('supertest')
const db = require('../data/dbConfig')
const server = require('./server')


beforeAll(async () => {
  await db.migrate.rollback()
  await db.migrate.latest()
}) // migrate
beforeEach(async () => {
  await db.seed.run()
}) // truncate and seed fresh data
afterAll(async () => {
  await db.destroy()
}) // disconnect from the db

describe('[GET] /resource', () => {
  test('gets all resources from the table', async () => {
    const res = await request(server).get('/resource')
    expect(res.body).toHaveLength(0)
    expect(res.body).toBeTruthy()
  })
  test('snapshot test', async () => {
    const res = await request(server).get('/resource')
    expect(res.body).toMatchSnapshot()
  })
})

describe('[POST] /resource', () => {
  test('responds with status code 201', async () => {
    const res = await request(server).post('/resource').send({
      name: 'Yellow Pages'
    })

    expect(res.status).toBe(201)
  })

  test('responds with newly created resource', async () => {
    const res = await request(server).post('/resource').send({
      name: 'Yellow Pages'
    })
    expect(res.body).toMatchObject({ id: 2, name: 'Yellow Pages'})
  })
})

describe('[DELETE] /resource', () => {
  test('responds with status code 200', async () => {
    
  })
})
