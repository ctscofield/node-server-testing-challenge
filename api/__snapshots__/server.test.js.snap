// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[GET] /resource snapshot test 1`] = `Array []`;
